src/main/java/TaskList.java

```diff
@@ -0,0 +1,109 @@
+//        this.MANAGER = manager;
+//
+//    }
+
+    public static void reportList() throws CustomException {
+        int len = MANAGER.size();
+        if (len != 0) {
+            for (int i = 0; i < len; i++) {
+                System.out.println((i + 1) + ". " + MANAGER.get(i).toString());
+            }
+        } else {
+            throw new CustomException("your task list is currently empty. " +
+                    "Add a task first:)");
+        }
+    }
+
+    public static Task findTask(int num) throws CustomException {
+        Task desiredTask = new Task("empty task");
+        int len = MANAGER.size();
+
+        if ((num <= len) && (len > 0) && (num > 0)) {
+            for (int i = 1; i <= len; i++) {
+                if (i == num) {
+                    desiredTask = MANAGER.get(i - 1);
+                }
+            }
+            return desiredTask;
+        } else if (num < 0) {
+            throw new CustomException("Invalid task ID: number must be a positive integer:)");
+        } else if (num > len) {
+            throw new CustomException("Invalid task ID: this task number does not exist as of now.");
+        } else { // length of task list == 0
+            throw new CustomException("your task list is currently empty. " +
+                    "Add a task first:)");
+        }
+    }
+
+    public static void addTask(String taskType, String instruction) throws CustomException {
+        Task toAdd = new Task("empty task");
+        if (taskType.equals("todo")) {
+            String description = instruction.substring(4);
+            if (description.isBlank()) {
+                throw new CustomException("todo description cannot be blank! Please add task details");
+            } else {
+                toAdd = new Todo(description.substring(1));
+            }
+        } else if (taskType.equals("deadline")) {
+            int index = instruction.indexOf("/by");
+            if (index != -1) {
+                String description = instruction.substring(9, index);
+                String time = instruction.substring(index + 4);
+                toAdd = new Deadline(description, time);
+            } else {
+                throw new CustomException("Incorrect time format: ensure to prefix time with '/by'");
+            }
+        } else {
+            int index = instruction.indexOf("/at");
+            if (index != -1) {
+                String description = instruction.substring(6, index);
+                String time = instruction.substring(index + 4);
+                int indexOfDifferentiator = time.indexOf("-");
+                String dateAndStartTime = time.substring(0, indexOfDifferentiator).trim();
+                String endTime = time.substring(indexOfDifferentiator + 1).trim();
+                toAdd = new Event(description, dateAndStartTime, endTime);
+            } else {
+                throw new CustomException("Incorrect time format: ensure to prefix time with '/at'");
+            }
```

Comment:
You could have used switch statements and component functions to make the code more modular.